scanpy/tools/_dendrogram.py

```diff
@@ -155,7 +155,11 @@ def dendrogram(
         )
 
     # aggregate values within categories using 'mean'
-    mean_df = rep_df.groupby(level=0, observed=True).mean()
+    mean_df = (
+        rep_df.groupby(level=0, observed=True)
+        .mean()
+        .loc[categories]  # Fixed ordering for pandas < 2
```

Comment:
oops, I missed that before. Maybe we should add a helper or `.ordered_categories` or something for the now three spots where this is used.

```suggestion
        .loc[
            self.categories_order
            if self.categories_order is not None
            else self.categories
        ]
```

Reply:
Any ideas for a test case for this change?

Reply:
Whenever `categories_order` is specified, that’s the order, and whenever it isn’t, the implicit order is used, no? So e.g. the test I fixed in 70e586698c4beda6808e15bc8091e4ce24509ffb

Reply:
But this change hasn't been committed, so I don't think that test could rely on this since it's passing right now.

Reply:
@flying-sheep 

> Seems like if we want to support pandas <2, we need to use that pattern or abstract it away somehow. Please do that.

Can you please suggest a test for this? I don't know which functions would be calling this as opposed the other location you fixed it in

Reply:
Ah, I understand now, the dendrogram defines the order, and plots happening after it use its order, so this is fine as is.